### PR TITLE
Run a validator (Polkadex)

### PR DESCRIPTION
**Pull request by frankli-dev**
User guide how to bond & how to run a validator (Polkadex) for Public testnet announcement.

<strong>I will make a new release (v0.4.0) after this PR is merged!</strong>

### CODE REVIEW

File: docs/run-a-validator.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+rustup update
+```
+
+If not, this command will fetch the latest version of Rust and install it.
+
+```
+curl https://sh.rustup.rs -sSf | sh -s -- -y
+```
+
+> If you do not have "curl" installed, run "sudo apt install curl"
```
Comment (simonsso) [inline docs/run-a-validator.md:46]: Put curl in the sudo apt-get install section

File: docs/run-a-validator.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+```
+rustc --version
+```
+
+Finally, run this command to install the necessary dependencies for compiling and running the Polkadex node.
+
+```
+sudo apt install make clang pkg-config libssl-dev build-essential
+```
```
Comment (simonsso) [inline docs/run-a-validator.md:64]: Move this before rustup

File: docs/run-a-validator.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+```
+git clone https://github.com/Polkadex-Substrate/Polkadex.git
+cd Polkadex
+```
+
+Build native code with the cargo release profile.
+
+```
+git checkout v0.4.0
```
Comment (simonsso) [inline docs/run-a-validator.md:106]: tag not yes released - known issue

File: docs/run-a-validator.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+./scripts/init.sh
```
Comment (simonsso) [inline docs/run-a-validator.md:107]: Must be updated only fetches wasm for nightly not nightly 20210511 which we are pinned to

**Comment by frankli-dev** [comment]
resolved all the above comments.

File: docs/run-a-validator.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+### Synchronize Chain Data
+
+Download `customSpecRaw.json` file for the Polkadex Public Testnet
+```
+cd $HOME
+curl -o customSpecRaw.json https://github.com/Polkadex-Substrate/Polkadex/releases/download/v0.4.0/customSpecRaw.json
+```
+
+You can begin syncing your node by running the following command:
```
Comment (pdexpab) [inline docs/run-a-validator.md:123]:
```suggestion
You can begin syncing your node by running the following command if you do not want to start in validator mode right away:
```

File: docs/run-a-validator.md [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+
+```
+$HOME/Polkadex/target/release/polkadex-node --chain=$HOME/customSpecRaw.json --bootnodes /ip4/13.235.92.50/tcp/30333/p2p/12D3KooWBRsL9KPkMeWxTMq5aSbgUWEMgwzWpWDA6EqQ6A2KTDoR --pruning=archive
+```
+
+if you do not want to start in validator mode right away.
```
Comment (pdexpab) [inline docs/run-a-validator.md:129]:
```suggestion

```

File: docs/run-a-validator.md [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+
+![Bonding Preferences](./screenshots/bonding_preferences.png)
+
+- <strong>Stash account</strong> - Select your Stash account. In this example, we will bond 100 PDEX, where the minimum bonding amount is 100. Make sure that your Stash account contains at least this much. You can, of course, stake more than this.
+- <strong>Controller account</strong> - Select the Controller account created earlier. This account will also need a small amount of PDEX in order to start and stop validating.
+- <strong>Value bonded</strong> - How much PDEX from the Stash account you want to bond/stake. Note that you do not need to bond all of the PDEX in that account. Also note that you can always bond more PDEX later. However, withdrawing any bonded amount requires the duration of the unbonding period. On Polkadex, the planned unbonding period is 28 days.
+- <strong>Payment destination</strong> - The account where the rewards from validating are sent. More info here. If you'd like to redirect payments to an account that is neither the controller nor the stash account, set one up. Please note that it is extremely unsafe to set an exchange deposit address as the recipient of the staking rewards.
```
Comment (pdexpab) [inline docs/run-a-validator.md:197]: "More info here." delete 
```suggestion
- <strong>Payment destination</strong> - The account where the rewards from validating are sent. If you'd like to redirect payments to an account that is neither the controller nor the stash account, set one up. Please note that it is extremely unsafe to set an exchange deposit address as the recipient of the staking rewards.
```

**Review by simonsso — APPROVED** [review]
We need to implement the twitter fund bot. for this to work Link will be updated here

**Review by simonsso — APPROVED** [review]

File: docs/run-a-validator.md [hunk 8]
```diff
@@ -47,10 +47,14 @@ If not, this command will fetch the latest version of Rust and install it.
 curl https://sh.rustup.rs -sSf | sh -s -- -y
 ```
 
-To configure your shell, run the following command.
+To to build you need to add some toolschains
```
Comment (pavankanteti) [inline docs/run-a-validator.md:50]: 'To' is used twice.

File: docs/run-a-validator.md [hunk 9]
```diff
@@ -176,7 +176,7 @@ The `--pruning=archive` flag is implied by the `--validator` flag, so it is only
 If you are interested in determining how much longer you have to go, your server logs (printed to STDOUT from the `Polkadex` process) will tell you the latest block your node has processed and verified. You can then compare that to the current highest block via [Telemetry](https://telemetry.polkadot.io/#list/Polkadex%20Public%20Testnet) or the [PolkadotJS Block Explorer](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fblockchain.polkadex.trade#/explorer).
 
 ## Bond PDEX
-
+[Create two accounts](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fblockchain.polkadex.trade#/accounts) [Here is an instruction to how](https://www.youtube.com/watch?v=hhUZ40ZWqkE) and transfer some PDEX tokens to the main account
```
Comment (pavankanteti) [inline docs/run-a-validator.md:180]: Here is an instruction to how >> Here are the instructions

File: docs/run-a-validator.md [hunk 10]
```diff
@@ -2,26 +2,26 @@
...
+- **customSpecRaw.json** https://github.com/Polkadex-Substrate/Polkadex/releases/download/v0.4.0/customSpecRaw.json
 
 ## Initial Set-up
 
 ### Requirements
 
-The most common way for a beginner to run a validator is on a cloud server running Linux. You may choose whatever VPS provider you prefer, and whichever operating system you are comfortable with. For this guide we will be using <strong>Ubuntu 20.04</strong>, but the instructions should be similar for other platforms.
+The most common way for a beginner to run a validator is on a cloud server running Linux. You may choose whatever VPS provider you prefer, and whichever operating system you are comfortable with. For this guide we will be using **Ubuntu 20.04**, but the instructions should be similar for other platforms.
 
 The transactions weights in Polkadot were benchmarked on standard hardware. It is recommended that validators run at least the standard hardware in order to ensure they are able to process all blocks in time. The following are not minimum requirements but if you decide to run with less than this, you may experience performance issues.
```
Comment (pavankanteti) [inline docs/run-a-validator.md:16]: Transaction weights

File: docs/run-a-validator.md [hunk 11]
```diff
@@ -191,10 +191,10 @@ First, go to the [Staking](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fblockch
...
-- <strong>Payment destination</strong> - The account where the rewards from validating are sent. If you'd like to redirect payments to an account that is neither the controller nor the stash account, set one up. Please note that it is extremely unsafe to set an exchange deposit address as the recipient of the staking rewards.
+- **Stash account** - Select your Stash account. In this example, we will bond 100 PDEX, where the minimum bonding amount is 100. Make sure that your Stash account contains at least this much. You can, of course, stake more than this.
+- **Controller account** - Select the Controller account created earlier. This account will also need a small amount of PDEX in order to start and stop validating.
+- **Value bonded** - How much PDEX from the Stash account you want to bond/stake. Note that you do not need to bond all of the PDEX in that account. Also note that you can always bond more PDEX later. However, withdrawing any bonded amount requires the duration of the unbonding period. On Polkadex, the planned unbonding period is 28 days.
```
Comment (pavankanteti) [inline docs/run-a-validator.md:196]: Question mark at the end of the first sentence

File: docs/run-a-validator.md (hunk 10 above)
Comment (simonsso) [reply docs/run-a-validator.md:16]:
```suggestion
The transaction weights in Polkadot were benchmarked on standard hardware. It is recommended that validators run at least the standard hardware in order to ensure they are able to process all blocks in time. The following are not minimum requirements but if you decide to run with less than this, you may experience performance issues.
```

File: docs/run-a-validator.md (hunk 8 above)
Comment (simonsso) [reply docs/run-a-validator.md:50]:
```suggestion
Add the required toolchains with rustup
```

File: docs/run-a-validator.md (hunk 9 above)
Comment (simonsso) [reply docs/run-a-validator.md:180]:
```suggestion
[Create two accounts](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fblockchain.polkadex.trade#/accounts) [Here is one instruction video how to create accounts] (https://www.youtube.com/watch?v=hhUZ40ZWqkE) and transfer some PDEX tokens to the main account.
```

File: docs/run-a-validator.md (hunk 11 above)
Comment (simonsso) [reply docs/run-a-validator.md:196]:
```suggestion
- **Value bonded** - How much PDEX from the Stash account you want to bond/stake? You do not need to bond all of the PDEX in that account. Also note that you can always bond more PDEX later. However, withdrawing any bonded amount requires the duration of the unbonding period. On Polkadex, the planned unbonding period is 28 days.
```

**Review by simonsso — APPROVED** [review]

**Review by simonsso — APPROVED** [review]

**Pull request merged**